drivers/modem/modem_cmd_handler.c

```diff
@@ -199,6 +223,29 @@ static struct modem_cmd *find_cmd_match(struct modem_cmd_handler_data *data)
 	return NULL;
 }
 
+static struct modem_cmd *find_cmd_direct_match(
+		struct modem_cmd_handler_data *data)
+{
+	int j, i;
+
+	for (j = 0; j < ARRAY_SIZE(data->cmds); j++) {
+		if (!data->cmds[j] || data->cmds_len[j] == 0U) {
+			continue;
+		}
+
+		for (i = 0; i < data->cmds_len[j]; i++) {
+			/* match start of cmd */
+			if (data->cmds[j][i].direct &&
+			    (strlen(data->cmds[j][i].cmd) == 0 ||
```

Comment:
Instead of calling `strlen()` to see if the string is empty, you could just do `data->cmds[j][i].cmd[0] == '\0'` here